### PR TITLE
refactor: move contact to account

### PR DESCRIPTION
**Pull request by nicolasburtey**

### CODE REVIEW

**Comment by dolcalmi** [comment]
as far as I remember contact was part of the user not accounts... did this change?

**Comment by nicolasburtey** [comment]
> as far as I remember contact was part of the user not accounts... did this change?

we're moving using username to `handle`, and those will be attached either to account or wallets. this is just a first PR is that direction.

File: src/app/accounts/update-contact-alias.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}): Promise<AccountContact | ApplicationError> => {
+  const repo = AccountsRepository()
+  const account = await repo.findById(accountId)
+  if (account instanceof Error) {
+    return account
+  }
+
+  const contact = account.contacts.find((contact) => contact.username === username)
+  if (!contact) {
+    return new ValidationError(`User doesn't have contact ${username}`)
```
Comment (bodymindarts) [inline src/app/accounts/update-contact-alias.ts:21]: `ValidationError` is just a base class and shouldn't be instantiated directly - please create a new error type.

File: src/app/accounts/update-contact-alias.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  }
+  contact.alias = alias as ContactAlias
```
Comment (bodymindarts) [inline src/app/accounts/update-contact-alias.ts:23]: does the alias names have any constraints (length / charset)? do we need a `checkedToContactAlias`?

File: src/app/users/add-new-contact.ts [hunk 3]
```diff
@@ -1,35 +1,35 @@
-import { UsersRepository } from "@services/mongoose"
+import { AccountsRepository } from "@services/mongoose"
 
 export const addNewContact = async ({
```
Comment (bodymindarts) [inline src/app/users/add-new-contact.ts:3]: file should be moved under `app/accounts` or?

File: src/app/accounts/update-contact-alias.ts (hunk 2 above)
Comment (nicolasburtey) [reply src/app/accounts/update-contact-alias.ts:23]: yup. good point. actually as of now this could be a source of ddos.

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**